src/lv_objx/lv_list.c

```diff
@@ -366,6 +367,36 @@ void lv_list_set_style(lv_obj_t * list, lv_list_style_t type, const lv_style_t *
     }
 }
 
+/**
+ * Set layout of a list
+ * @param list pointer to a list object
+ * @param layout which layout should be used
+ */
+ void lv_list_set_layout(lv_obj_t * list, lv_list_layout_t layout)
+ {
+    lv_list_ext_t * ext = lv_obj_get_ext_attr(list);
+	
+	/* Update list layout if necessary */
+	if (layout != lv_list_get_layout(list)) {
+
+		/* Get the first button on the list */
+		lv_obj_t * btn = lv_list_get_prev_btn(list, NULL);
+		
+		/* Visit all buttons on the list and update their layout  */
+        while(btn != NULL) {
+			if (LV_LIST_LAYOUT_HOR == layout) {
+				lv_btn_set_fit2(list, LV_FIT_FLOOD, LV_FIT_TIGHT);
+			} else { /* LV_LIST_LAYOUT_VER */
+				lv_btn_set_fit(list, LV_FIT_TIGHT);
+			}
+
+            btn = lv_list_get_prev_btn(list, btn);
+        }
+
```

Comment:
It seems you forget to apply the layout with `lv_page_set_scr_layout(list, LV_LAYOUT_COL_M/ROW_M)`